feat: add a new highlighter component

Pratiyankkumar opened the pull request:
Added a new Highlighter component which highlights the text as it were highlighted by a human .
Additional features include: 
1)  Colour of highlighting : We can provide the colour of  highlighting as a prop.
2)  Action of highlighter: We can provide the action of  highlighter as a prop , which can be a circle or highlighter.

Preview shown in the image :
![Screenshot 2025-02-11 203134](https://github.com/user-attachments/assets/597f60a5-21d3-448c-a277-4d7f8ae30412)

Pratiyankkumar commented:
@dillionverma Appreciate you taking a look! 👀 Let me know if it's good to merge.

dillionverma commented:
Great component idea, I could see it being useful for a lot of cases.

However, I think the PR in it's current state feels incomplete.

Some points to improve:

1. Expose access to all underlying props https://github.com/rough-stuff/rough-notation. Looks like there are more annotation types but this PR limits it to only 2. There are also several other useful props which are hidden.
2. Give more demo examples

Pratiyankkumar commented:
ok will do that, thanks for pointing out

Pratiyankkumar commented:
> Great component idea, I could see it being useful for a lot of cases.
> 
> However, I think the PR in it's current state feels incomplete.
> 
> Some points to improve:
> 
> 1. Expose access to all underlying props https://github.com/rough-stuff/rough-notation. Looks like there are more annotation types but this PR limits it to only 2. There are also several other useful props which are hidden.
> 2. Give more demo examples

Added more annotations ✅
Added other useful props to give full access of the  component to the user ✅
Given more demo examples ✅
here are the changes : 

https://github.com/user-attachments/assets/fd3c8cf5-f146-4793-a236-e77d3bd05529

Pratiyankkumar commented:
@dillionverma Please tell me if its good to merge or something else is remaining

itsarghyadas commented:
We apologize for the delay. We will review it as soon as we have time and get back to you. Thank you.
